tools/autograd/derivatives.yaml

```diff
@@ -972,7 +972,12 @@
   save_invstd: not_implemented("native_batch_norm_backward save_invstd")
 
 - name: native_layer_norm(Tensor input, int[] normalized_shape, Tensor? weight, Tensor? bias, float eps) -> (Tensor, Tensor, Tensor)
-  input, weight, bias: "GradMode::is_enabled() || grads[1].defined() || grads[2].defined() ? infinitely_differentiable_native_layer_norm_backward(grads[0], grads[1], grads[2], input, result1, result2, weight, normalized_shape, eps, grad_input_mask) : (grads[0].defined() ? native_layer_norm_backward(grads[0].is_contiguous() ? grads[0] : grads[0].contiguous(), input, normalized_shape, result1, result2, weight, bias, grad_input_mask) : std::tuple<Tensor, Tensor, Tensor>())"
+  input, weight, bias: "GradMode::is_enabled() || grads[1].defined() || grads[2].defined() ? infinitely_differentiable_native_layer_norm_backward(grads[0], grads[1], grads[2], input, result1, result2, weight, normalized_shape, eps, grad_input_mask) : (grads[0].defined() ? native_layer_norm_backward(grads[0], input, normalized_shape, result1, result2, weight, bias, grad_input_mask) : std::tuple<Tensor, Tensor, Tensor>())"
+
+- name: native_layer_norm_backward(Tensor grad_out, Tensor input, int[] normalized_shape, Tensor mean, Tensor rstd, Tensor? weight, Tensor? bias, bool[3] output_mask) -> (Tensor, Tensor, Tensor)
```

Comment:
I'm not sure how this can ever be called as the code above is explicitly using `infinitely_differentiable_native_layer_norm_backward` if GradMode is enabled (meaning that you're preparing for double backward).

Reply:
Same as the comment above. In jit, autodiff doesn't use autograd logic. So in the case of double backward, first backward call would use `native_layer_norm_backward` as the backward function defined in `symbolic_script.cpp` (https://github.com/pytorch/pytorch/pull/50467/files#diff-e690d44b82baeb013fa244ea3ceb4b2b9bf7a9b7b525723818652186e6af2468R1156)

The second backward function kicks in and it will be looking for a backward function for `native_layer_norm_backward`

Reply:
That sounds bad no?
If we have an efficient implementation of the backward that is infinitely differentiable, we should use that for both autograd.

Reply:
Note that using both of them will also allow the existing gradcheck for the vanilla autograd to also check this implementation to make sure that it is correct.

Reply:
The meta test case also does grad & gradgrad check, which in the jitted version covers the added code path.

> That sounds bad no?
It does sounds bad. The alternative here is to remove `infinitely_differentiable_native_layer_norm_backward` and just use `native_layer_norm_backward`.

Reply:
> The alternative here is to remove infinitely_differentiable_native_layer_norm_backward and just use native_layer_norm_backward.

Yes! Code removal 🎉